cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java

```diff
@@ -1179,4 +1092,54 @@ protected RegistryClient getRegistryClient()
         return new RegistryClient();
     }
 
+    /**
+     * Lookup the Service URL for the given standard.  The current AuthMethod
+     * will be taken into account.
+     *
+     * @param standard  The URI standard to look up.
+     * @return          URL for the service.
+     * @throws AccessControlException       If the URL cannot be found for the
+     *                                      provided AuthMethod.
+     */
+    private URL lookupServiceURL(final URI standard)
+            throws AccessControlException
+    {
+        final URL serviceURL = getRegistryClient()
+                .getServiceURL(this.serviceID, standard, getAuthMethod());
+
+        if (serviceURL == null)
+        {
```

Comment:
(this goes with the comment below on getAuthMethod())  -  I'd throw a general RuntimeException here because the service doesn't have a capability the client is expecting....